wordpress-mobile/WordPress-Android
# Jetpack Focus: show overlay on Jetpack feature card tap

Closes #17336

This PR hooks in the showing of the overlay view on tap of the dashboard card.
The logic is handled separately from the “screen_type” enum because this particular overlay did not fit that pattern. I am open to changing if the reviewer feels this would be a better implementation.


Light | Dark
-- | --
<img width="250" height="500" alt="Light" src="https://user-images.githubusercontent.com/506707/210891143-b2e80545-0b2d-4bbd-ba70-1958b061986c.png"> | <img width="250" height="500" alt="Dark" src="https://user-images.githubusercontent.com/506707/210891160-8e3585bf-18a5-4d42-b939-e9d06a63ff85.png">


The following existing events will be tracked with “source=card”.
remove_feature_overlay_displayed
remove_feature_overlay_dismissed
remove_feature_overlay_link_tapped
remove_feature_overlay_button_tapped

Notes:
I ran into an issue with the overlay not showing on subsequent taps of the card, this was due to `JetpackFeatureFullScreenOverlayFragment` using `ActivityViewModels`. I have switched this out to `viewModels`. It was definitely a head scratcher. I didn’t see any instances in which the viewModel was being shared, but please call me out if this is not the case.

**To test:**
- Install WP app
- Login
- Navigate to Me -> App Settings -> Privacy Settings and turn on Collection Information
- Navigate to -> Debug Settings -> and Enable jp_removal_three and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown
- Tap on the card
- ✅ Verify the overlay is displayed
- ✅ Verify the log contains 🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"three","source":"card"}
- Tap on Continue with Jetpack
- ✅ Verify the overlay closes
- ✅ Verify the log contains 🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"three","source":"card","dismissal_type":"continue"}
- Tap on the card
- Tap on Learn more at Jetpack.com link
- ✅ Verify the in-app browser launches
- Tap back to return to the app
- ✅ Verify the log contains 🔵 Tracked: remove_feature_overlay_link_tapped, Properties: {"phase":"three","source":"card"}
- ✅ Verify the overlay is not showing
- Tap on the card
- Tap on the "Switch to the Jetpack app" button
- ✅ Verify the log contains 🔵 Tracked: remove_feature_overlay_button_tapped, Properties: {"phase":"three","source":"card"}

## Regression Notes
1. Potential unintended areas of impact
The overlay is not shown on tap

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Update existing where needed

PR submission checklist:
